apps/apputil.cpp

```diff
@@ -77,29 +77,85 @@ int inet_pton(int af, const char * src, void * dst)
 }
 #endif // _WIN32 && !HAVE_INET_PTON
 
-sockaddr_in CreateAddrInet(const string& name, unsigned short port)
+#define NI_MAXNUMERICHOST 64
+
+sockaddr_any CreateAddrInet(const string& name, const string& port)
 {
-    sockaddr_in sa;
-    memset(&sa, 0, sizeof sa);
-    sa.sin_family = AF_INET;
-    sa.sin_port = htons(port);
+    int error = 0;
+    const char *description;
+    struct addrinfo *destination;
+    struct addrinfo hints;
+    size_t length;
+    char ni[NI_MAXNUMERICHOST];
+    sockaddr_any sa;
+    sa.reset();
+    // sa.sin6_family = AF_INET6;
+    // sa.sin6_port = htons(port);
 
     if ( name != "" )
     {
-        if ( inet_pton(AF_INET, name.c_str(), &sa.sin_addr) == 1 )
-            return sa;
-
-        // XXX RACY!!! Use getaddrinfo() instead. Check portability.
-        // Windows/Linux declare it.
-        // See:
-        //  http://www.winsocketdotnetworkprogramming.com/winsock2programming/winsock2advancedInternet3b.html
-        hostent* he = gethostbyname(name.c_str());
-        if ( !he || he->h_addrtype != AF_INET )
-            throw invalid_argument("CreateAddrInet: host not found: " + name);
-
-        sa.sin_addr = *(in_addr*)he->h_addr_list[0];
-    }
+        memset (&hints, 0, sizeof hints);
+        hints.ai_family = AF_UNSPEC;    // Allow IPv4 or IPv6
+        hints.ai_socktype = SOCK_DGRAM; // Datagram socket
+        hints.ai_flags = 0;
+        hints.ai_protocol = 0;          // Any protocol
+
+        error = getaddrinfo(name.c_str(), port.c_str(), &hints, &destination);
```

Comment:
A question: did you intend to support specifications like `myhost.com:complex-main` (as `myhost.com:5000`) ? Then this syntax is likely not supported by `UriParser`.